src/python/pants/pantsd/service/scheduler_service.py

```diff
@@ -89,11 +103,19 @@ def _enqueue_fs_event(self, event):
                       .format(len(event['files']), event['subscription']))
     self._event_queue.put(event)
 
-  def _maybe_invalidate_scheduler_batch(self, files):
-    invalidating_files = self._invalidating_files
-    if any(f in invalidating_files for f in files):
-      self._logger.fatal('saw file events covered by invalidation globs, terminating the daemon.')
-      self.terminate()
+  def _maybe_invalidate_scheduler_batch(self, changed_globs):
```

Comment:
So, I think that this method can be made much, much simpler.

Since `product_request` caches the request, I think that this entire method can just be:
```
new_snapshot = self._get_snapshot(..)
if new_snapshot.directory_digest != self._invalidating_snapshot.directory_digest:
  # self.terminate() and return, as below
```